### ISSUE TITLE
Model.from_db_value() signature change in Django 2.0

### ISSUE THREAD

**Issue by mongkyo** [django-oscar/django-oscar#3045]
Hello.
I tried to apply django-oscar to my project.
But I found the following warning messages:  “RemovedInDjango30Warning: : Remove the context parameter from NullCharField.from_db_value() and UppercaseCharField.from_db_value(). Support for it will be removed in Django 3.0.
   RemovedInDjango30Warning”

Seeing warning messages every time I run tests are by no means good, so I wonder if you are open to contributions regarding fixing this. If so, I will be glad to help.

### Reproduce

1. I think that is the cause of the error is the  `UppercaseCharField` used in `oscar/models/fields/__init__.py` of from_db_value function.
And also `NullCharField` is the cause of the error used in `oscar/models/fields/__init__.py` of from_db_value function.

2.
```
class NullCharField(CharField):
    """
    CharField that stores '' as None and returns None as ''
    Useful when using unique=True and forms. Implies null==blank==True.

    Django's CharField stores '' as None, but does not return None as ''.
    """
    description = "CharField that stores '' as None and returns None as ''"

    def __init__(self, *args, **kwargs):
        if not kwargs.get('null', True) or not kwargs.get('blank', True):
            raise ImproperlyConfigured(
                "NullCharField implies null==blank==True")
        kwargs['null'] = kwargs['blank'] = True
        super().__init__(*args, **kwargs)

    def contribute_to_class(self, cls, name, **kwargs):
        super().contribute_to_class(cls, name, **kwargs)
        setattr(cls, self.name, Creator(self))

    def from_db_value(self, value, expression, connection, context):
        value = self.to_python(value)
        # If the value was stored as null, return empty string instead
        return value if value is not None else ''
```
```
class UppercaseCharField(CharField):
    """
    A simple subclass of ``django.db.models.fields.CharField`` that
    restricts all text to be uppercase.

    Defined with the with_metaclass helper so that to_python is called
    https://docs.djangoproject.com/en/1.6/howto/custom-model-fields/#the-subfieldbase-metaclass  # NOQA
    """

    def contribute_to_class(self, cls, name, **kwargs):
        super().contribute_to_class(
            cls, name, **kwargs)
        setattr(cls, self.name, Creator(self))

    def from_db_value(self, value, expression, connection, context):
        return self.to_python(value)

    def to_python(self, value):
        val = super().to_python(value)
        if isinstance(val, str):
            return val.upper()
        else:
            return val
```

### Technical details

Python   3.7.3
Django   2.2
pytest   4.4.0
django-oscar   2.0b1 (edited)

**Comment by solarissmoke** [comment]
Thanks for reporting this. We would be grateful if you are able to make a PR to fix this!

**Comment by solarissmoke** [comment]
Fixed in #3046
